### ISSUE TITLE
Change imageIdentifier later?

### ISSUE THREAD

**Issue by agordeev** [twitter/ios-twitter-image-pipeline#16]
Is it possible to change `imageIdentifier` later somehow?

Use case: the user uploads an image to the server. Before sending the image to the server, the app stores it into the cache. After the image is uploaded, it retrieves an url of the image and updates `imageIdentifier`.

Or may be there's a better way to handle such use case?

**Comment by NSProgrammer** [comment]
This is a good use case and something we should consider better support for in TIP itself.  At the moment, we will store an image to a pipeline and upon completion (and knowing the canonical URL) will save the same image to the new identifier while deleting the old identifier.

I think the addition of a method on the pipeline to simply change the identifier is what you are looking for and would serve your needs (and ours too!).

```objc
- (void)changeIdentifierForImageWithIdentifier:(NSString *)currentIdentifier toIdentifier:(NSString *)newIdentifier;
```

We will consider this for a future drop of TIP.  In the meantime, you should be able to unblock yourself by either using the same pattern we have or by forking TIP to add this method yourself.

**Comment by NSProgrammer** [comment]
This has been merged with v2.4.4 - closing
